driver/remote/driver.go

```diff
@@ -23,17 +23,39 @@ type tlsOpts struct {
 }
 
 func (d *Driver) Bootstrap(ctx context.Context, l progress.Logger) error {
-	return nil
+	for i := 0; ; i++ {
+		info, err := d.Info(ctx)
+		if err != nil {
+			return err
+		}
+		if info.Status != driver.Inactive {
+			return nil
+		}
+
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			if i > 10 {
+				i = 10
+			}
+			time.Sleep(time.Duration(i) * time.Second)
+		}
+	}
 }
 
 func (d *Driver) Info(ctx context.Context) (*driver.Info, error) {
 	c, err := d.Client(ctx)
 	if err != nil {
-		return nil, errors.Wrapf(driver.ErrNotConnecting, err.Error())
+		return &driver.Info{
```

Comment:
I think this should still return error and we should check for "network error" on the caller side.

Reply:
I agree about preserving the error to be able to display - I'm not sure about how much to change the caller side. `Info` gets called before `Boot`, which requires some fiddly logic changes in generic code to ignore network errors in that case.

I think there's two scenarios for errors here:
- Failure to connect to the endpoint providing the driver (can't connect to kubernetes, can't connect to docker daemon, etc) - this kind of failure means it's not worth trying to bootstrap, something is wrong.
- Failure to connect to the individual driver node (as in this case) - bootstrapping here is fine

Could we maybe rework status to be a struct and add an `Err` field to it? Then we could express a status of a node that's `inactive` because of an `err`. It currently feels like the error returned from here is quite tied to it's status, and is a little inconsistent across the drivers.